blog/_posts/2020-03-31-extensions.md

```diff
@@ -0,0 +1,52 @@
+---
+slug: more-customizable-tiltfiles-with-extensions
+date: 2020-04-01
+author: victor
+layout: blog
+title: "More customizable Tiltfiles with Extensions"
+subtitle: "Share Tiltfile code snippets with the Tilt community in a streamlined platform. And get a free t-shirt!"
+image: "/assets/images/more-customizable-tiltfiles-with-extensions/michael-dziedzic-XTblNijO9IE-unsplash.jpg"
+image_caption: "Photo by <a href='https://unsplash.com/@lazycreekimages?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Michael Dziedzic</a> on <a href='https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Unsplash</a>"
+tags:
+  - tilt
+keywords:
+  - extensions
+  - tilt
+  - tiltfile
+---
+
+Tilt lets you declare your dev environment in Starlark, a subset of Python. But Starlark is missing a big feature! There's no package manager. Starlark doesn't have built-in ways to share your configuration.
+
+The Tilt community noticed! We've seen people:
+- Copy and paste "best practices" for different languages
+- Struggle to share Tiltfiles across multiple repos
+- Send each other Tiltfile snippets
+
+So we've created Tilt extensions as a first baby step toward a multi-service dev package manager to address this gap.
+
+Tilt extensions are available starting in [v0.12.11](https://github.com/windmilleng/tilt/releases/tag/v0.12.11). [Upgrade](https://docs.tilt.dev/upgrade.html) to the latest version of Tilt.
+
+## What's an extension, exactly? And how do you use it?
+An extension is a packaged function that you import and call, all right inside your Tiltfile. 
```

Comment:
You can also import another file locally using `load()`, "right inside your Tiltfile." That's not the point. The point is to do so remotely, and more importantly, to be able to use functionality you didn't write yourself (since ideally a lot of people will be sharing these.)

Without making these points, the 14-year-old edgelord nerd in me reads this whole section and replies with, "and why should I care?"

Reply:
Re-worded to includes these two points. Great observation.